### DISCUSSION TITLE
Support PTZ actions in camera entities

### DISCUSSION THREAD

**Discussion by tsvi** [home-assistant/architecture#508] [Entity Models]
As discussed in the comments of #153, certain cameras support PTZ actions.

Dafang hacks for Xiaomi cameras as an example allows movement by sending MQTT actions, at the moment it is implemented as creating 2 covers: one for left/right, another for up/down.

I think we should add this as a general optional support for cameras. We can use the services that have been implemented for the [ONVIF cameras](home-assistant/home-assistant#11630), I'd also like to surface the controls to the UI.

_Originally posted by @tsvi in https://github.com/home-assistant/architecture/issues/153#issuecomment-465888366_

See also home-assistant/home-assistant#11630

**Comment by tsvi** [comment]
@dgomes https://github.com/home-assistant/architecture/issues/153#issuecomment-465972724

**Comment by dgomes** [comment]
I said a new component, but it can be the same component (camera) with added methods.
- pan()
- tilt()
- reset_position()

**Comment by rohankapoorcom** [comment]
I support this. I think there's a few more methods needed:

* go_to_preset()
    * A lot of cameras support presets that you can save on the camera and then it moves to that saved point.
* zoom() 
* patrol() 
    * Patrol makes the camera automatically move along a preset path, going from position to position.

**Comment by Kane610** [comment]
Is there a reason to support this outside of the camera component? If there is no camera there is no reason for PTZ

**Comment by dgomes** [comment]
There are lights with ptz (like the ones you find in clubs)

Yeh... you usually don't find those at home ;) I'm fine with extending the camera component.

**Comment by Gamester17** [comment]
There are pan & tilt mounts that you can buy and use separate from the camera. They are still meant to be used with a camera but they are not integrated as in PTZ cameras. That could be one reason to keep this outside the camera component, or at least keep in mind that they are available as separate parts.

Examples:
- https://www.ebay.co.uk/itm/312492920874
- https://www.ebay.com/itm/Electric-Bracket-RS485-Control-Pan-Tilt-Security-CCTV-Camera-Mount-12V-DC-Power-/391829180915
- https://www.ebay.co.uk/itm/PT-Mini-Pan-tilt-Assembled-w-Servo-Camera-Mount-Platform-Arduino-FPV-/233093862901

**Comment by Kane610** [comment]
That's an implementation detail
